osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/AlarmTypeRegisterLookup.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.alarm;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.DlmsObjectType;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.AlarmTypeDto;
+
+/**
+ * Gives the position of the alarm code as indicated by the AlarmType in the bit string
+ * representation of the alarm register.
+ *
+ * <p>A position of 0 means the least significant bit, up to the maximum of 31 for the most
+ * significant bit. Since the 4 most significant bits in the object are not used according to the
+ * DSMR documentation, the practical meaningful most significant bit is bit 27.
+ */
+@Getter
+public enum AlarmTypeRegisterLookup {
```

Comment:
Have you considered using a nested EnumMap `EnumMap<DlmsObjectType, EnumMap<AlarmTypeDto,Integer>>` as an alternative?

Reply:
Yes, i did, but wanted to move these values out of the class AlarmHelperService, to make it look cleaner.